### ISSUE TITLE
rejectVersionFunction gets the semver string as its first argument, not the name of the package

### ISSUE THREAD

**Issue by MissingNoIOI** [raineorshine/npm-check-updates#1442]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)

---

## Steps to Reproduce

.ncurc.js:

```js
module.exports = {
  rejectVersion: (name, semver) => {
    console.log(name)
    if (name == 'chai' && parseInt(semver[0]?.major) >= 5) {
      return true
    }
    if (name == 'sinon-chai' && parseInt(semver[0]?.major) >= 4) {
      return true
    }
    if (name == 'chai-exclude' && parseInt(semver[0]?.major) >= 3) {
      return true
    }
    if (name == 'sort-keys' && parseInt(semver[0]?.major) >= 5) {
      return true
    }
    if (name == '@typescript-eslint/parser' && parseInt(semver[0]?.major) >= 7) {
      return true
    }
    if (name == '@typescript-eslint/eslint-plugin' && parseInt(semver[0]?.major) >= 7) {
      return true
    }
    return false
  }
}
```

Steps:

## Current Behavior
The first argument that gets passed in is the string of the version, like "^4.3.16"

## Expected Behavior
The first argument that gets passed in is the name of the package, as described in the documentation. This way the rejectVersion function is not really useful

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

Yes, that's definitely not right. I'll work on cleaning that up. Any change in behavior will have to wait for a major version update, but in the mean time can you try switching to `filter` instead of `filterVersion`? I believe it passes `(name, semver)` as expected.

**Comment by MissingNoIOI** [comment]
Yeah, with `reject` it inputs the name correctly, but the semver is now the current version, instead of the one to be upgraded to, so I cant filter the version either.

**Comment by raineorshine** [comment]
`filter` and `reject` run before the latest versions are fetched, so you will not have access to the upgraded versions there. They are preferred when filtering based on name or current version since they reduce the number of HTTP requests.

[filterResults](https://github.com/raineorshine/npm-check-updates#filterresults) runs after the latest versions are fetched, allowing you to filter on the upgraded version.

**Comment by raineorshine** [comment]
Thanks, I will fix it!

**Comment by XhmikosR** [comment]
Is this still valid?

**Comment by raineorshine** [comment]
I suggest that we either deprecate or simply remove the function usage of `filterVersion` and `rejectVersion` so that `filter` and `reject` become the canonical, pre-fetch filtering mechanisms. They pass both named and version anyway.

Keep `filterVersion` and `rejectVersion` support for the other types (string, wildcard, glob, comma-or-space-delimited list, /regex/)

What do you think?

**Comment by XhmikosR** [comment]
Agreed. I say we better drop it since we are in a major bump.
